frontend/app_flowy/lib/user/presentation/folder/folder_widget.dart

```diff
@@ -0,0 +1,252 @@
+  var page = _FolderPage.options;
+
+  @override
+  Widget build(BuildContext context) {
+    return SizedBox(
+      height: 250,
+      child: _mapIndexToWidget(context),
+    );
+  }
+
+  Widget _mapIndexToWidget(BuildContext context) {
+    switch (page) {
+      case _FolderPage.options:
+        return FolderOptionsWidget(
+          onPressedCreate: () {
+            setState(() => page = _FolderPage.create);
+          },
+          onPressedOpen: () {
+            _openFolder();
+          },
+        );
+      case _FolderPage.create:
+        return CreateFolderWidget(
+          onPressedBack: () {
+            setState(() => page = _FolderPage.options);
+          },
+          onPressedCreate: widget.createFolderCallback,
+        );
+      case _FolderPage.open:
+        break;
+      default:
+    }
+    return Container();
+  }
+
+  Future<void> _openFolder() async {
+    final directory = await FilePicker.platform.getDirectoryPath();
+    if (directory != null) {
+      await getIt<SettingsLocationCubit>().setLocation(directory);
+      await widget.createFolderCallback();
+    }
+  }
+}
+
+class FolderOptionsWidget extends StatelessWidget {
+  const FolderOptionsWidget({
+    Key? key,
+    required this.onPressedCreate,
+    required this.onPressedOpen,
+  }) : super(key: key);
+
+  final VoidCallback onPressedCreate;
+  final VoidCallback onPressedOpen;
+
+  @override
+  Widget build(BuildContext context) {
+    return ListView(
+      shrinkWrap: true,
+      children: <Widget>[
+        Card(
+          child: ListTile(
+            title: FlowyText.medium(
+              LocaleKeys.settings_files_createNewFolder.tr(),
+            ),
+            subtitle: FlowyText.regular(
+              LocaleKeys.settings_files_createNewFolderDesc.tr(),
+            ),
+            trailing: _buildTextButton(
+              context,
+              LocaleKeys.settings_files_create.tr(),
+              onPressedCreate,
+            ),
+          ),
+        ),
+        Card(
+          child: ListTile(
+            title: FlowyText.medium(
+              LocaleKeys.settings_files_openFolder.tr(),
+            ),
+            subtitle: FlowyText.regular(
+              LocaleKeys.settings_files_openFolderDesc.tr(),
+            ),
+            trailing: _buildTextButton(
+              context,
+              LocaleKeys.settings_files_open.tr(),
+              onPressedOpen,
+            ),
+          ),
+        ),
+      ],
+    );
+  }
+}
+
+class CreateFolderWidget extends StatefulWidget {
+  const CreateFolderWidget({
+    Key? key,
+    required this.onPressedBack,
+    required this.onPressedCreate,
+  }) : super(key: key);
+
+  final VoidCallback onPressedBack;
+  final Future<void> Function() onPressedCreate;
+
+  @override
+  State<CreateFolderWidget> createState() => CreateFolderWidgetState();
+}
+
+@visibleForTesting
+class CreateFolderWidgetState extends State<CreateFolderWidget> {
+  var _folderName = 'appflowy';
```

Comment:
localization?

Reply:
No. It's a default folder name, not a display text. I assume that we should `appflowy` as a default folder name or something like `document`?